omegaconf/listconfig.py

```diff
@@ -504,6 +504,16 @@ def __contains__(self, item: Any) -> bool:
         return False
 
     def _set_value(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> None:
+        try:
+            previous_content = self.__dict__["_content"]
```

Comment:
interesting that there is no need for a shallow copy here (unlike the DictConfig). how come?

Reply:
`self.__dict__["_content"] = {}`allocates a new pointer thus making `previous_content` able to be assigned the last content state without needing a copy. I forgot to ask you if I should copy this behaviour in dictconfig in this same diff.

Reply:
Not sure I follow what behavior.
maybe separate PR to clean DictConfig's shallow copy?

Reply:
Okay